File: app/src/main/java/com/hfad/mymovies/viewmodels/SearchFragmentViewModel.java
package com.hfad.mymovies.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.hfad.mymovies.data.models.MovieDiscover;
import com.hfad.mymovies.data.MovieRepository;

import java.util.List;
import java.util.Locale;

public class SearchFragmentViewModel extends ViewModel {

    private MovieRepository mMovieRepository;
    private String language = Locale.getDefault().getLanguage();

    public SearchFragmentViewModel(MovieRepository movieRepository) {
        mMovieRepository = movieRepository;
    }

    public LiveData<List<MovieDiscover>> searchMovie(String name){
        return mMovieRepository.searchMovie(name, language);
    }
}

File: app/src/main/java/com/hfad/mymovies/viewmodels/SearchFragmentViewModel.kt
package com.hfad.mymovies.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.hfad.mymovies.data.MovieRepository
import com.hfad.mymovies.data.models.MovieDiscover
import java.util.*

class SearchFragmentViewModel(private val movieRepository: MovieRepository) : ViewModel() {
    private val language = Locale.getDefault().language
    fun searchMovie(name: String?): LiveData<List<MovieDiscover>> {
        return movieRepository.searchMovie(name, language)
    }
}